Allow the use of font awesome with configuration

cesarizu opened the pull request:
This PR implements a way to allow users of the addon to specify easily that they want to use font-awesome icons instead of glyphicons.

balinterdi commented:
That's a great idea.

One thing I would make simpler (I think :) ), is now that you added the possibility of passing in full and empty star names, I would not use the config and just pass in class names directly:

```handlebars
{{star-rating stars=stars ... starFullClassNames="fa fa-star" starEmptyClassNames="fa fa-star-o"}}
```

If we do like this, we don't need any of the configuration code, the extra `ember-get-config` dependency, and the overridden `init` method.

In the case of a component, inserted directly into the DOM, I also prefer to see how the element will look like (what classes it will have) just by looking at the component's template and possibly the component's js.

I think it'd be a bit unintuitive to have to look at the config to be able to see what classes are added and it would also make it global for the whole application (since it's set in the config and not on a case-by-case basis).

What do you think?

balinterdi commented:
I would also call the classes `fullClasses` and `emptyClasses` for brevity (as `star` is already implied).

cesarizu commented:
I updated this PR to remove the configuration options and added some other ways to do that in the README file. I changed the `star*ClassNames` to `*ClassNames`, leaving the `Names` there to be consistent with the base component `classNames` attribute.

cesarizu commented:
Do you think it's a good idea to add another predefined component `star-rating-fa` to this addon?

This might be a bit more involed: maybe try to detect if there's a bootstrap addon installed to use glyphicons or check if there's a font-awesome addon to use font-awesome from the component?

balinterdi commented:
Hey César,

That's perfect as it is, I only made minor tweaks to a couple of texts.

I wouldn't create a `star-rating-fa` component as inheriting from `star-rating` and then overriding the classes gets the job done, as you nicely described in the README. Although having components out-of-the-box for major icon factories (glyphicon, font awesome, any others) might be a good idea, so that this addon really becomes an off-the-shelf widget. Let me dwell on it for a bit.

I'm going to cut a new release, thanks a bunch for your cooperation.

cesarizu commented:
No problem. If I can help with anything just let me know.

balinterdi commented:
@cesarizu If you'd like to help with #10 , that would be great. Thank you!
